extensions/notification/queueProcessor/QueueProcessor.js

```diff
@@ -0,0 +1,133 @@
+
+const BackbeatConsumer = require('../../../lib/BackbeatConsumer');
+
+class QueueProcessor extends EventEmitter {
+
+    /**
+     * Create a queue processor object to activate notification from a
+     * kafka topic dedicated to dispatch messages to a destination/target.
+     *
+     * @constructor
+     * @param {Object} kafkaConfig - kafka configuration object
+     * @param {string} kafkaConfig.hosts - list of kafka brokers
+     *   as "host:port[,host:port...]"
+     * @param {Object} notifConfig - notification configuration object
+     * @param {String} notifConfig.topic - notification topic name
+     * @param {String} notifConfig.queueProcessor - config object
+     *   specific to queue processor
+     * @param {String} notifConfig.queueProcessor.groupId - kafka
+     *   consumer group ID
+     * @param {String} notifConfig.queueProcessor.retryTimeoutS -
+     *   number of seconds before giving up retries of an entry
+     * @param {String} destinationType - destination type
+     */
+    constructor(kafkaConfig, notifConfig, destinationType) {
+        super();
+        this.kafkaConfig = kafkaConfig;
+        this.notifConfig = notifConfig;
+        this._consumer = null;
+        this._destinationType = destinationType;
+        this._destination = null;
+
+        this.logger = new Logger('Backbeat:Notification:QueueProcessor');
+    }
+
+    _getDestinationConfiguration() {
+        // TODO: util function to fetch destination configuration from a file
+        // check if this has to be moved to destination class.
+        return undefined;
+    }
+
+    _setupDestination(destinationType, done) {
+        // get the destination configuration
+        // get destination object using the destination type
+        // set the destination object to this._destination
+        // this._destination.start();
+        done();
+    }
+
+    /**
+     * Start kafka consumer. Emits a 'ready' even when consumer is ready.
+     *
+     * Note: for tests, with auto.create.topics.enable option set on
+     * kafka container, this will also pre-create the topic.
+     *
+     * @param {object} [options] options object
+     * @param {boolean} [options.disableConsumer] - true to disable
+     *   startup of consumer (for testing: one has to call
+     *   processQueueEntry() explicitly)
+     * @return {undefined}
+     */
+    start(options) {
+        this._setupDestination(this.destinationType, err => {
+            if (err) {
+                this.logger.info('error setting up destination',
+                    { error: err.message });
+                return undefined;
+            }
+            if (options && options.disableConsumer) {
+                this.emit('ready');
+                return undefined;
+            }
+            const groupId =
+                `${this.notifConfig.queueProcessor.groupId}-${this.site}`;
+            const concurrency = this.notifConfig.queueProcessor.concurrency;
```

Comment:
You can use object deconstruction `const { groupId, concurrency } = this.notifConfig.queueProcessor` to shorten the code.